radix-engine/src/state_manager/staging.rs

```diff
@@ -1,189 +1,581 @@
-use crate::ledger::*;
+use crate::transaction::TransactionReceipt;
 use crate::types::*;
-use radix_engine_interface::api::types::SubstateId;
+use crate::{ledger::*, transaction::TransactionResult};
 
-/// Nodes form an acyclic graph towards the parent
-struct StagedSubstateStoreNode {
```

Comment:
Can we retain old SubstateManager in order to be able to compare/benchmark between the two implementations?

Reply:
Sure, should I keep it under same name and rename the new one? Also it's not really quite the same since this one is basically an immutable store that supports branching natively while the old implementation doesn't.

Reply:
We could just rename the old one to something like "DeprecatedStagingStore"